packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -37,17 +38,26 @@ import {
 import { Custodian } from "../plugins";
 import { TokenType } from "../token";
 
+type GasFeeTransactionParams = {
+    maxFeePerGas?: BigNumberish;
+    maxPriorityFeePerGas?: BigNumberish;
+};
+
 export class EVMAAWallet<B extends EVMBlockchainIncludingTestnet = EVMBlockchainIncludingTestnet> {
     private sessionKeySignerAddress?: Hex;
     private crossmintService: CrossmintWalletService;
     private publicClient: PublicClient;
     private ecdsaValidator: KernelValidator<entryPoint, "ECDSAValidator">;
-    private account: KernelSmartAccount<EntryPoint, HttpTransport, TChain>;
-    private kernelClient: KernelAccountClient<
-        entryPoint,
-        HttpTransport,
-        TChain,
-        KernelSmartAccount<entryPoint, HttpTransport, TChain>
+    // account represents the one got from eth_requestAccounts.
+    // we need to expose it because createKernelAccountClient returns that it can be undefined, and some methods require it to be specifically included because of that.
+    public account: KernelSmartAccount<EntryPoint, HttpTransport, TChain>;
```

Comment:
this part I don't understand. I think we should keep this private if there's any way to do so as this is the kind of internal we want to hide from the developer

Reply:
ah ok, sorry for missing some context 
I see that this is for [this change](https://github.com/Crossmint/crossmint-sdk/pull/481/files#diff-74abb96fec665aa4e65410adc9d1d80a34750de5bd7e3a01222827ee21c0217fR20)

To make sure I understand, what's happening is that:
With a non-eip1559-supported chain (astar), `signTypedData` doesn't work unless you use different parameters (include `account`)
If that's the case, would we be able to leave `account` private by doing this?
```
async signTypedData(params: TypedDataDefinition) {
    try {
        return await this.kernelClient.signTypedData({ ...params, account: this.account });
    } catch (error) {
        logError("[SIGN_TYPED_DATA] - ERROR", {
            service: SCW_SERVICE,
            error: errorToJSON(error),
            signer: this.kernelClient.account,
        });
        throw new Error(`Error signing typed data. If this error persists, please contact support.`);
    }
}

Reply:
and then `reservoirAdapter` doesn't have to change?